packages/dashboard-provider/lib/DashboardProvider.ts

```diff
@@ -0,0 +1,149 @@
+import type {
+  JSONRPCRequestPayload,
+  JSONRPCErrorCallback,
+  JSONRPCResponsePayload
+} from "ethereum-protocol";
+import { callbackify } from "util";
+import WebSocket from "isomorphic-ws";
+import delay from "delay";
+import open from "open";
+import {
+  sendAndAwait,
+  createMessage,
+  connectToMessageBusWithRetries,
+  getMessageBusPorts,
+  base64ToJson,
+  LogMessage
+} from "@truffle/dashboard-message-bus";
+import { startDashboardInBackground } from "@truffle/dashboard";
+import { timeout } from "promise-timeout";
+import { DashboardProviderOptions } from "./types";
+import debugModule from "debug";
+
+export class DashboardProvider {
```

Comment:
Should we make this support EIP-1193 while we're at it?

Reply:
We definitely could, but given that we don't want to promote the `DashboardProvider` as the "main" way to interact with the dashboard (and instead want to promote using `truffle dashboard` + RPC URL), I'm not sure if it adds too much. Wouldn't hurt to have of course, but it's maybe not worth the effort at this point.

Reply:
Agreed